torchci/components/DarkModeEChart.tsx

```diff
@@ -0,0 +1,193 @@
+import ReactECharts from "echarts-for-react";
+import { useDarkMode } from "lib/DarkModeContext";
+import { useEffect, useRef } from "react";
+
+export default function DarkModeEChart(props: any) {
+  const { darkMode } = useDarkMode();
+  const chartRef = useRef(null);
+
+  // Function to update all text colors in the chart SVG
+  const updateChartColors = () => {
+    if (!chartRef.current) return;
+
+    // Access the DOM elements directly
+    const chartElement = chartRef.current as any;
+    if (!chartElement || !chartElement.getEchartsInstance) return;
+
+    try {
+      const echartsInstance = chartElement.getEchartsInstance();
+      const dom = echartsInstance.getDom();
+      if (!dom) return;
+
+      const textColor = darkMode ? "#E0E0E0" : "#212529";
+
+      // Update all text elements
+      dom.querySelectorAll("text").forEach((text: SVGTextElement) => {
+        text.style.fill = textColor;
+        text.setAttribute("fill", textColor);
+      });
+
+      // Special handling for title text
+      dom.querySelectorAll(".ec-title, .ec-title-sub").forEach((el: any) => {
+        if (el && el.style) {
+          el.style.fill = textColor;
+          el.style.color = textColor;
+        }
+
+        // Also look for text elements inside titles
+        el.querySelectorAll("text, tspan").forEach((text: SVGTextElement) => {
+          text.style.fill = textColor;
+          text.setAttribute("fill", textColor);
+        });
+      });
+
+      // Update SVG title and subtitle elements
+      dom
+        .querySelectorAll(
```

Comment:
Nit (unless it violates some linter rules)
```suggestion
      dom.querySelectorAll(
```